crypto/evp/evp_enc.c

```diff
@@ -605,6 +862,22 @@ int EVP_CIPHER_CTX_set_padding(EVP_CIPHER_CTX *ctx, int pad)
         ctx->flags &= ~EVP_CIPH_NO_PADDING;
     else
         ctx->flags |= EVP_CIPH_NO_PADDING;
+
+    if (ctx->cipher != NULL && ctx->cipher->prov != NULL) {
+        const OSSL_PARAM params[] = {
+            OSSL_PARAM_int(OSSL_CIPHER_PARAM_PADDING, &pad),
+            OSSL_PARAM_END
```

Comment:
Perhaps not for this, but I'd prefer to see the END macro take an empty arglist.  Or rather, is it really needed, just ```{NULL}``` seems better.  Like Andy used to say, avoid creating a "dialect of C" via the pre-processor.

Reply:
Not for this PR.

Reply:
#8704